zeebe/gateway-protocol/src/main/proto/rest-api.yaml

```
@@ -1936,6 +1937,176 @@ paths:
         "500":
           $ref: "#/components/responses/InternalServerError"
 
+  /roles:
+    post:
+      tags:
+        - Role
+      operationId: createRole
+      summary: Create role
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: "#/components/schemas/RoleCreateRequest"
+      responses:
+        "202":
```

🔧 Shouldn't this be a `201 Created`? I guess we have a similar inconsistency for other endpoints that "create" resources. While a 200 is understandable maybe, a 202 usually means the action was just accepted but not completed yet. This isn't really the case here, is it?

```suggestion
        "201":
```

---

I was following `createUser`, which does return 202. As far as I understand, the API is only eventually consistent (because writes need to be exported to ES/OS), so 202 would make sense. 201 also works for me though, or even 200 if we make it return the full entity.

---

Yeah, I saw the create user endpoint as well. The eventual consistency does not count here, if you ask me. The command is processed in the engine and comes back with a result that is also written to the log at that point. The eventual consistency is only for secondary storage, that should not influence the API response.
So, a 201 makes most sense here in my regard.

If you're up for it, you can also adjust the user endpoint, actually.

---

Either here or in a follow-up to not bloat this PR

---

I'll make a follow-up PR.